docs-src/self-hosted/how-to-set-up-codec-server.md

```diff
@@ -8,53 +8,41 @@ tags:
   - how-to
 ---
 
-Use a Codec Server to decode your encoded [payloads](/concepts/what-is-a-payload) and integrate it with the Temporal Web UI and CLI commands when debugging your Workflows.
+Use a Codec Server to programmatically decode your encoded [payloads](/concepts/what-is-a-payload).
 
-A Codec Server is an HTTP or HTTPS Server that you create and host.
-It must be configured to use a [Payload Codec](/concepts/what-is-a-payload-codec) with the required decode logic and encryption keys.
-Temporal Cloud requires an HTTPS Codec Server.
+A Codec Server is an HTTP server that uses your custom Codec logic to decode your data remotely.
+The Codec Server is independent of the Temporal Cluster and decodes your encrypted payloads through predefined endpoints. You create, operate, and manage access to your Codec Server in your own environment.
+The `temporal` CLI and the Web UI in turn provide built-in hooks to call the Codec Server to decode encrypted payloads on demand.
 
 The Codec Server is independent of the Temporal Server and decodes your encrypted payloads through endpoints.
-When you set the codec endpoint in the Temporal Web UI, the Web UI uses the remote endpoint to send encoded payloads to the Codec Server and receive decoded payloads from the Codec Server.
+When you configure a Codec Server endpoint in the Temporal Web UI or CLI, the Web UI and CLI use the remote endpoint to receive decoded payloads from the Codec Server.
```

Comment:
Same noun thing as earlier. "CLI" feels like you're referring to Unix rather than the `temporal` or `tctl` utility but you know what? I don't think anyone will misunderstand and if you don't add it, I think it's just fine.